src/main.rs

```diff
@@ -138,9 +155,20 @@ fn main(opts: Opts) -> Result<(), Error> {
             break;
         }
 
-        if let Ok(event) = watcher_rx.try_recv() {
-            debug!("received a watcher event {:#?}", event);
-            info!("reload configuration");
+        // retrieve all pending events from watch
+        let watch_event_count = watcher_rx
+            .try_iter()
+            .map(|event| {
+                debug!("received a watch event '{:#?}'", event);
+                event
+            })
+            .count();
+
+        if watch_event_count > 0 {
+            info!(
```

Comment:
Should be debug level rather than info one?

Reply:
Not sure, I found it useful to have a log during reload.